dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/trackedentityfilter/TrackedEntityFilterControllerTest.java

```diff
@@ -0,0 +1,247 @@
+    assertEquals(HttpStatus.OK, response.status());
+  }
+
+  @Test
+  void shouldFailWhenAssignedUserModeProvidedAndNoAssignedUsers() {
+    HttpResponse response =
+        POST(
+            "/trackedEntityInstanceFilters",
+            """
+                {
+                  "program": {"id": "%s"},
+                  "entityQueryCriteria": {
+                    "assignedUserMode": "PROVIDED"
+                  },
+                  "name": "name"
+                }
+            """
+                .formatted(programId));
+
+    assertEquals(HttpStatus.CONFLICT, response.status());
+    assertContains(
+        "Assigned Users cannot be empty with PROVIDED assigned user mode",
+        response.error().getMessage());
+  }
+
+  @Test
+  void shouldFailWhenOrgUnitModeSelectedAndNoOrgUnitProvided() {
+    HttpResponse response =
+        POST(
+            "/trackedEntityInstanceFilters",
+            """
+                {
+                  "program": {"id": "%s"},
+                  "entityQueryCriteria": {
+                    "ouMode": "SELECTED"
+                  },
+                  "name": "name"
+                }
+            """
+                .formatted(programId));
+
+    assertEquals(HttpStatus.CONFLICT, response.status());
+    assertContains(
+        "Organisation Unit cannot be empty with SELECTED org unit mode",
+        response.error().getMessage());
+  }
+
+  @Test
+  void shouldFailWhenNonExistentAttributeProvided() {
+    UID nonExistentAttribute = UID.generate();
+
+    HttpResponse response =
+        POST(
+            "/trackedEntityInstanceFilters",
+            """
+                {
+                  "program": {"id": "%s"},
+                  "entityQueryCriteria": {
+                    "attributeValueFilters": [{"eq": "value", "attribute": "%s"}]
+                  },
+                  "name": "name"
+                }
+            """
+                .formatted(programId, nonExistentAttribute.getValue()));
+
+    assertEquals(HttpStatus.CONFLICT, response.status());
+    assertContains("No tracked entity attribute found", response.error().getMessage());
+  }
+
+  @Test
+  void shouldCreateTrackedEntityFilterWhenNullOperatorSupplied() {
+    String operatorFilterId = createOperatorFilter("null operator filter", "null", true);
+
+    JsonTrackedEntityFilter trackedEntityFilter =
+        GET("/trackedEntityInstanceFilters/{id}", operatorFilterId)
+            .content()
+            .as(JsonTrackedEntityFilter.class);
+
+    assertFalse(trackedEntityFilter.isEmpty());
+    assertEquals(programId, trackedEntityFilter.getProgram());
+    JsonObject attributeObject =
+        trackedEntityFilter.getEntityQueryCriteria().getAttributeValueFilters().getObject(0);
+    assertEquals(trackedEntityAttributeId, attributeObject.getString("attribute").string());
+    assertEquals(Boolean.TRUE, attributeObject.getBoolean("null").booleanValue());
+  }
+
+  @Test
```

Comment:
Would it make sense to add a test case when the value for the null operator is `null`?
The creation should fail, right?

Reply:
if the value is `null` it doesn't fail, and, regardless of the operator, it will only store the attribute.
as an example:

`"attributeValueFilters": [{"eq": null, "attribute": "ruQQnf6rswq"}]`
`"attributeValueFilters": [{"null": null, "attribute": "ruQQnf6rswq"}]`

both store only `{"attributeValueFilters": [{"attribute": "ruQQnf6rswq"}]}`